go/rpcclientlib/viewing_key_client.go

```diff
@@ -220,10 +219,13 @@ func (c *ViewingKeyClient) addFromAddressToCallParamsIfMissing(method string, ar
 		// We only modify `eth_call` requests where the `from` field is not set.
 		return args, nil
 	}
+
+	if c.viewingKeyAddr == (common.Address{}) {
```

Comment:
This check was missing before. Best to check this, to not accidentally leave the from field as nil and send a wasted request.